operator/controllers/reconcilers/server/statefulset_reconciler.go

```diff
@@ -206,8 +206,8 @@ const (
 )
 
 func (s *ServerStatefulSetReconciler) GetConditions() []*apis.Condition {
-	ready := s.StatefulSet.Status.ReadyReplicas >= s.StatefulSet.Status.Replicas
-	s.Logger.Info("Checking conditions for stateful set", "ready", ready, "replicas", s.StatefulSet.Status.Replicas, "availableReplicas", s.StatefulSet.Status.AvailableReplicas)
+	ready := s.StatefulSet.Status.ReadyReplicas >= *s.StatefulSet.Spec.Replicas
+	s.Logger.Info("Checking conditions for stateful set", "ready", ready, "replicas", *s.StatefulSet.Spec.Replicas, "availableReplicas", s.StatefulSet.Status.AvailableReplicas)
```

Comment:
perhaps add `s.StatefulSet.Status.Replicas` in logging?

Reply:
I feel like we should add a nil pointer check here to prevent from unwanted panic?

Reply:
@abhimanyu003  The * here derefs `Replicas`, which is a pointer. In our code that is always set (either explicitly or as a default), so I didn't check it. However, I'll add the check so that it doesn't cause confusion and makes that constraint explicit.

s is also a pointer, but it is the function pointer receiver. In the codebase, we typically don't check this for nil because we follow the pattern

```
s, err := NewStatefulSetReconciler(....)
if (err != nil) {
    // handle error and return
}
....
// use s, which is not nil
```